specs-go/config.go

```diff
@@ -126,7 +126,7 @@ type Hook struct {
 	Path    string   `json:"path"`
 	Args    []string `json:"args,omitempty"`
 	Env     []string `json:"env,omitempty"`
-	Timeout *int     `json:"timeout,omitempty"`
```

Comment:
> Add back the pointer to this field so we can validate a actual user provided 0 value.

This seems like it would be an issue with the string properties de-pointerized by #653.  I think that validation tools should be using the JSON Schema or `interface{}` parsing to check this sort of thing (pull request to get that to happen in opencontainers/runtime-tools#197).  If the goal is to be able to distinguish “unset” from “set to an invalid/no-op value”, then we do want a pointer here, but probably also want to get the pointer back for [`CgroupsPath`][1], etc.  Or is there a reason why we want to be able to easily identify “`timeout` set to 0” in Go but don't need that for “`cgroupsPath` set to an empty string”?

[1]: https://github.com/opencontainers/runtime-spec/pull/653/files#diff-7f24d60f0cbb9c433e165467e3d34838L142